pilot/pkg/serviceregistry/external/conversion.go

```diff
@@ -152,7 +152,7 @@ func convertEndpoint(service *model.Service, servicePort *networking.Port,
 			},
 			LbWeight: endpoint.Weight,
 			Labels:   endpoint.Labels,
-			TLSMode:  tlsMode,
+			TLSMode:  model.GetTLSModeFromEndpointLabels(endpoint.Labels),
```

Comment:
I think this is not correct. We will service account check that is being done in `convertWorkloadInstances` before calling `convertEndpoint`

Reply:
to keep consistent with pod, we should not treat it as mtls when sa present.

Only when labels set, it is mtls.

Reply:
I see. Then why don't we do retain the previous change ` model.GetTLSMode` without considering SA and use it for both POD and SE?

Reply:
Is the current name `GetTLSModeFromEndpointLabels` confusing to you?

Reply:
It is not for the current usage. But I am looking at reusing it for both pod and here as you have done previously. I think that will allow us to be consistent in logic across both

Reply:
I see nevermind. `PodTLSMode` also is using the same function before also.

Reply:
I dont thikn we can derive just from labels. According to the docs, we treat the presence of service account on a workload entry as a way of indicating that the workload has a Sidecar. Otherwise, users have to add the mtls label to all workload entries (not to mention, remember the mtls mode). Put another way, there is no point in having the service account on a workload entry if there is going to be no sidecar as the workloadEntry is pointing to a VM.

Reply:
Got it.